g2p/transducer/__init__.py

```diff
@@ -645,3 +691,25 @@ def __call__(self, to_convert: str):
                 non_word_tg = TransductionGraph(token["text"])
                 tg += non_word_tg
         return tg
+
+    def check(self, tg: TransductionGraph, shallow=False, display_warnings=False):
+        # The obvious implementation fails, because we need to check only the words, not
+        # the text between the words!
+        # return self._transducer.check(tg) # <- complains about characters between words
+
+        # So, sadly, we redo the work of transduction so we can check the words only, step
```

Comment:
Yes, I suppose we do need to tokenize and transduce here. Can't think of a way around it.